### DISCUSSION TITLE
#[debug_handler] Doesnt work with FromRef<S>

### DISCUSSION THREAD

**Discussion by genusistimelord** [tokio-rs/axum#1354] [Q&A] [answered]
## Bug Report
`#[debug_handler]` displays
 ```
the trait bound `axum_flash::Config: axum::extract::FromRef<()>` is not satisfied
the following other types implement trait `axum::extract::FromRequest<S, B, M>`:
  (T1, T2)
  (T1, T2, T3)
  (T1, T2, T3, T4)
  (T1, T2, T3, T4, T5)
  (T1, T2, T3, T4, T5, T6)
  (T1, T2, T3, T4, T5, T6, T7)
  (T1, T2, T3, T4, T5, T6, T7, T8)
  (T1, T2, T3, T4, T5, T6, T7, T8, T9)
and 22 others
required for `extensions::state::FullState` to implement `axum::extract::FromRequest<(), axum::body::Body, axum_core::extract::private::ViaParts>`
```

### Description
when `#[debug_handler]` is enabled for a async function that has a FromRequestParts Built Full State that needs a requirement

```
where
    S: ::std::marker::Send + ::std::marker::Sync,
    axum_flash::Config: axum::extract::FromRef<S>,
```
    
    
It will Error out displaying this error message
If I comment it out the source will not Error and will build and run perfectly fine.

### Version
Axum 0.6.0-rc.1

### Platform
Windows 11

### Crates
AxumFlash:
This affects me when I need to pass axum_flash::Config as a State but building a FullState using FromRequest or FromRequestParts.

**Comment by jplatte** [comment, marked as answer]
See https://docs.rs/axum-macros/0.3.0-rc.1/axum_macros/attr.debug_handler.html#changing-state-type

**Reply by davidpdrsn** [reply]
Yep. Looks like `#[debug_handler(state = axum_flash::Config)]` should do it.

**Reply by genusistimelord** [reply]
thank you yeah that was my fault I looked at the Documents but forgot to switch it to the 0.3.0-rc.1 docs. since it always defaults to the last none RC version.
